q2_fmt/tests/test_engraftment.py

```diff
@@ -909,3 +915,125 @@ def test_sample_id_match(self):
                          time_column="group",
                          reference_column="Ref",
                          subject_column="subject")
+
+    def test_group_column_categorical(self):
```

Comment:
With the helper method refactor mentioned above, you won't need all of these individual column/type tests either - you can just have one that confirms the behavior given a specific column and column type.